### ISSUE TITLE
Building 32-bit compat runtimes on x86_64 fails

### ISSUE THREAD

**Issue by berolinux** [llvm/llvm-project#56941]
Building 32-bit compat runtimes on an x86_64 linux system fails with a likely incorrect error message:

```
CMake Error at /home/bero/abf/llvm/BUILD/llvm-project-release-15.x/libunwind/src/CMakeLists.txt:109 (message):
  Compiler doesn't support generation of unwind tables if exception support
  is disabled.  Building libunwind DSO with runtime dependency on C++ ABI
  library is not supported.
```

I'm not disabling exceptions or anything - so the real reason is likely elsewhere. Looking at runtime-bins/CMakeFiles/CMakeErrors.log, I see various tests failing with

```
[2/2] Linking C executable cmTC_6d9b4
FAILED: cmTC_6d9b4
: && /home/bero/abf/llvm/BUILD/llvm-project-release-15.x/xc --target=i686-openmandriva-linux-gnu -resource-dir=/home/bero/abf/llvm/BUILD/llvm-project-release-15.x/build32/./lib/clang/15.0.0 -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-comment -Wstring-conversion -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections  -nostdlib++ -nostdinc++ -nostdlib++ -Werror=unknown-pragmas -resource-dir=/home/bero/abf/llvm/BUILD/llvm-project-release-15.x/build32/./lib/clang/15.0.0 -Wl,--color-diagnostics CMakeFiles/cmTC_6d9b4.dir/src.c.o -o cmTC_6d9b4  -lc  -lNOTFOUND && :
clang-14: warning: argument unused during compilation: '-nostdinc++' [-Wunused-command-line-argument]
ld.lld: error: unable to find library -lNOTFOUND
```

(Note the `-lNOTFOUND`).

Looks like I'm missing the 32-bit version of some new dependency library (building 32-bit compat libs worked with 14.0.6), and instead of erroring out/warning about that, the build system adds -lNOTFOUND and complains later.

FWIW the `xc` script referenced in my output is just a wrapper calling `clang -m32`.
